### DISCUSSION TITLE
Kill-Switch feature

### DISCUSSION THREAD

**Discussion by Kyoshim** [wiresock/WireSockUI#49] [Q&A] [answered]
Hello,

Is there a kill-switch feature like the one on the Wireguard app ?

![image](https://github.com/wiresock/WireSockUI/assets/5623682/6816cef4-253b-49c0-8435-e1a62ce1d7db)

Thanks

**Comment by wiresock** [comment, marked as answer]
The kill-switch feature of Wireguard, as you’ve noted, operates by manipulating the routing table, a key component in network routing. This process adjusts the routing table to direct internet traffic flow, ensuring that it only passes through the VPN tunnel. When the kill switch is activated, the routing table is altered to route all traffic (0.0.0.0/0) via the VPN. Conversely, when the kill switch is off, the routing rules become more specific, often dividing the IP range into two segments (0.0.0.0/1 and 128.0.0.0/1) to manage traffic flow in a distinct manner. These alterations are observable when inspecting the routing table as the kill switch is toggled on and off. Similarly, the WireSock VPN client, when in virtual adapter mode, is expected to exhibit comparable behavior, routing all traffic through the VPN tunnel unless specific traffic is explicitly set to bypass it.

**Reply by Kyoshim** [reply]
Hello,

Thanks a lot for this proper answer !

Have a nice day :)
